onnxmltools/convert/lightgbm/operator_converters/LightGbm.py

```diff
@@ -147,118 +237,223 @@ def convert_lightgbm(scope, operator, container):
+                gbm_text['objective']))
+
     # Use the same algorithm to parse the tree
     for i, tree in enumerate(gbm_text['tree_info']):
         tree_id = i
         class_id = tree_id % n_classes
-        learning_rate = 1. # tree['shrinkage'] --> LightGbm provides figures with it already.
-        _parse_tree_structure(tree_id, class_id, learning_rate, tree['tree_structure'], attrs)
-
-    # Sort nodes_* attributes. For one tree, its node indexes should appear in an ascent order in nodes_nodeids. Nodes
-    # from a tree with a smaller tree index should appear before trees with larger indexes in nodes_nodeids.
+        # tree['shrinkage'] --> LightGbm provides figures with it already.
+        learning_rate = 1.
+        _parse_tree_structure(
+            tree_id, class_id, learning_rate, tree['tree_structure'], attrs)
+
+    # Sort nodes_* attributes. For one tree, its node indexes
+    # should appear in an ascent order in nodes_nodeids. Nodes
+    # from a tree with a smaller tree index should appear
+    # before trees with larger indexes in nodes_nodeids.
     node_numbers_per_tree = Counter(attrs['nodes_treeids'])
     tree_number = len(node_numbers_per_tree.keys())
     accumulated_node_numbers = [0] * tree_number
     for i in range(1, tree_number):
-        accumulated_node_numbers[i] = (accumulated_node_numbers[i - 1]
-                                       + node_numbers_per_tree[i - 1])
+        accumulated_node_numbers[i] = (
+            accumulated_node_numbers[i - 1] + node_numbers_per_tree[i - 1])
     global_node_indexes = []
     for i in range(len(attrs['nodes_nodeids'])):
         tree_id = attrs['nodes_treeids'][i]
         node_id = attrs['nodes_nodeids'][i]
-        global_node_indexes.append(accumulated_node_numbers[tree_id] + node_id)
+        global_node_indexes.append(
+            accumulated_node_numbers[tree_id] + node_id)
     for k, v in attrs.items():
         if k.startswith('nodes_'):
-            merged_indexes = zip(copy.deepcopy(global_node_indexes), v)
-            sorted_list = [pair[1] for pair in sorted(merged_indexes, key=lambda x: x[0])]
+            merged_indexes = zip(
+                copy.deepcopy(global_node_indexes), v)
+            sorted_list = [pair[1]
+                           for pair in sorted(merged_indexes,
+                                              key=lambda x: x[0])]
             attrs[k] = sorted_list
 
     # Create ONNX object
-    if (gbm_text['objective'].startswith('binary')
-            or gbm_text['objective'].startswith('multiclass')):
-        # Prepare label information for both of TreeEnsembleClassifier and ZipMap
+    if (gbm_text['objective'].startswith('binary') or
+            gbm_text['objective'].startswith('multiclass')):
+        # Prepare label information for both of TreeEnsembleClassifier
         class_type = onnx_proto.TensorProto.STRING
-        zipmap_attrs = {'name': scope.get_unique_variable_name('ZipMap')}
```

Comment:
If the model was generated for opset 7/8, please keep the old behavior.

Reply:
Done.